lib/Expr/AssignmentGenerator.cpp

```diff
@@ -331,7 +331,11 @@ AssignmentGenerator::getIndexedValue(const std::vector<unsigned char> &c_val,
                                      const unsigned int size) {
   std::vector<unsigned char> toReturn;
   const llvm::APInt index_val = index.getAPValue();
+#if LLVM_VERSION_CODE >= LLVM_VERSION(5, 0)
+  assert(!index_val.isSignMask() && "Negative index");
```

Comment:
I think you should use `index_val.isSignBitSet()` here.

Reply:
Thanks to @kren1 , he pushed me towards reading the commit message as well.
Your changes make sense.